grpclb/src/main/java/io/grpc/grpclb/GrpclbLoadBalancer2.java

```diff
@@ -133,7 +132,7 @@ public PickResult pickSubchannel(Attributes affinity, Metadata headers) {
   GrpclbLoadBalancer2(Helper helper, Factory pickFirstBalancerFactory,
       Factory roundRobinBalancerFactory) {
     this.helper = checkNotNull(helper, "helper");
-    this.serviceName = checkNotNull(helper.getAuthority(), "helper returns null authority");
+    checkNotNull(helper.getAuthority(), "helper returns null authority");
```

Comment:
Can you instead replace all where calls `helper.getAuthority()` with `serviceName`?